R/recipe_read_depends.R

```diff
@@ -0,0 +1,165 @@
+    if (!is.null(el$draft)) {
+      msg <- c("Using 'draft:' within an ordery.yml is deprecated and",
+               "will be removed in a future version of orderly.  Please",
+               "use the 'use_draft' argument to control draft usage.",
+               "If you want to use a recent version of a report that you",
+               'are developing simultaneously, use_draft = "newer"',
+               "will probably do what you want.")
+      orderly_warning(flow_text(msg))
+      assert_scalar_logical(el$draft,
+                            sprintf("%s:depends:%s:draft", filename, name))
+    } else {
+      el$draft <- NA
+    }
+
+    assert_named(el$use, TRUE, sprintf("%s:depends:%s:use", filename, name))
+    err <- !vlapply(el$use, function(x) is.character(x) && length(x) == 1)
+    if (any(err)) {
+      stop(sprintf("%s:depends:%s:use must all be scalar character",
+                   filename, name))
+    }
+
+    el$filename <- list_to_character(el$use, FALSE)
+    el$as <- names(el$use)
+    el$use <- NULL
+    el$is_pinned <- el$id != "latest"
+    el$index <- i
+
+    ## Bit of a faff here to get the format into something that will
+    ## serialise and interrogate nicely.
+    as_data_frame(el)
+  }
+
+  rbind_df(lapply(seq_along(x), check_depends1))
+}
+
+
+resolve_dependencies <- function(depends, config, use_draft, remote) {
+  assert_is(config, "orderly_config")
+  if (is.null(depends)) {
+    return(NULL)
+  }
+
+  if (!is.null(remote)) {
+    if (use_draft) {
+      stop("Can't use 'use_draft' with remote")
+    }
+    remote <- get_remote(remote, config)
+  }
+
+  depends_split <- unname(split(depends, depends$index))
+  for (i in seq_along(depends_split)) {
+    name <- depends_split[[i]]$name[[1]]
+    id <- depends_split[[i]]$id[[1]]
+    filename <- depends_split[[i]]$filename
+    if (is.null(remote)) {
+      ## This is ugly but needs to stay for another couple of versions
+      use_draft_i <- depends_split[[i]]$draft[[1]]
+      use_draft_i <- if (is.na(use_draft_i)) use_draft else use_draft_i
+      res <- resolve_dependencies_local(id, name, config, use_draft_i)
+    } else {
+      res <- resolve_dependencies_remote(id, name, config, remote)
+    }
+    info <- resolve_dependencies_validate(id, name, res$path, filename)
+    extra <- as_data_frame(c(res, info))
+    depends_split[[i]] <- cbind(depends_split[[i]], extra)
+  }
+
+  rbind_df(depends_split)
+}
+
+
+resolve_dependencies_local <- function(id, name, config, use_draft) {
+  path <- orderly_find_report(id, name, config, draft = use_draft,
+                              must_work = TRUE)
+  is_latest <- id == "latest" ||
+    basename(path) == orderly_latest(name, config, draft = use_draft)
+  list(path = path, is_latest = is_latest)
+}
+
+
+resolve_dependencies_remote <- function(id, name, config, remote) {
+  remote <- get_remote(remote, config)
+
+  versions <- remote$list_versions(name)
+  latest_version <- versions[length(versions)]
```

Comment:
You don't use `latest_version` below - maybe change to `versions[[length(versions)]]` and reuse on 114 and 120?